src/browser/provider/index.ts

```diff
@@ -381,15 +385,18 @@ export default class BrowserProvider {
         return await this.plugin.canResizeWindowToDimensions(browserId, width, height);
     }
 
-    public async maximizeWindow (browserId: string): Promise<void> {
+    public async maximizeWindow (browserId: string, isNativeAutomation: boolean): Promise<void> {
         const canUseDefaultWindowActions = await this.canUseDefaultWindowActions(browserId);
         const customActionsInfo          = await this.hasCustomActionForBrowser(browserId);
         const hasCustomMaximizeWindow    = customActionsInfo.hasMaximizeWindow;
 
+        if (canUseDefaultWindowActions && !hasCustomMaximizeWindow && isNativeAutomation)
+            return await this.plugin.maximizeWindowNativeAutomation(browserId);
+
         if (canUseDefaultWindowActions && !hasCustomMaximizeWindow)
             return await this._maximizeLocalBrowserWindow(browserId);
 
-        return await this.plugin.maximizeWindow(browserId);
+        return await this.plugin.maximizeWindowHeadless(browserId);
```

Comment:
this is not clear why we use the `maximizeWindowHeadless` method here.
I cannot see any hits or condition checks that we need to use `headless` here.
Perhaps, there is a problem with method name.

Reply:
Now it looks very confusing:
```
if (nativeAutomation) 
    this.plugin.maximize()
else
    this.maximizeHeadless();
```